### PR TITLE
SList & Better DHV Toy

### PR DESCRIPTION
**Pull request by Kroronos**
Scrumbles Frames includes basic implementation of the SList (that is any list that appears in our Wireframes).
Features: 
Title
Sorting (Works badly with numbers unless single digits have a 0 preappended)
Scrollbar
Import List, Adding Items, Clearing List, Appending List, Deleting Item That Is Selected (No way to currently test this)

Improved developerHomeView by adding the three SLists that appear in the wireframes.
There are some issues with width for longer entries atm but this should be fixed when I allow horizontal expansion (haven't done so yet because other elements need to be added in that horizontal space), if this continues the easiest fix will be to add horizontal scrollbar.

### CODE REVIEW

File: source/developerHomeView.py [hunk 1]
```diff
@@ -8,4 +9,20 @@ def __init__(self, parent, controller,user):
...
+        self.usernameLabel.pack()
+
+        self.productBacklogList = ScrumblesFrames.SList(self, "PRODUCT BACKLOG")
+        self.teamMemberList = ScrumblesFrames.SList(self, "TEAM MEMBERS")
+        self.assignedItemList = ScrumblesFrames.SList(self, "ASSIGNED ITEMS")
+
+        backlog = ["GUI", "Styling File", "Hashing", "Encrypted Database", "Global Macros"]
+        teamMember = ["Anthony", "Kathy", "Liu", "Dave", "Josh", "Kayla"]
+        assignedItem = ["Processing", "Statistics", "UML Diagram", "Use Cases", "Refactor", "User Stories"]
+
```
Comment (fallinbryan) [inline source/developerHomeView.py]: https://github.com/CEN3031-group16/GroupProject/wiki/Scrumbles-API-Documentation#get-data-example

File: source/developerHomeView.py [hunk 2]
```diff
@@ -15,12 +15,20 @@ def __init__(self, parent, controller,user):
         self.teamMemberList = ScrumblesFrames.SList(self, "TEAM MEMBERS")
         self.assignedItemList = ScrumblesFrames.SList(self, "ASSIGNED ITEMS")
 
-        backlog = ["GUI", "Styling File", "Hashing", "Encrypted Database", "Global Macros"]
-        teamMember = ["Anthony", "Kathy", "Liu", "Dave", "Josh", "Kayla"]
-        assignedItem = ["Processing", "Statistics", "UML Diagram", "Use Cases", "Refactor", "User Stories"]
+        controller.dataConnection.connect()
+        backlog = ScrumblesData.ScrumblesData.getData(controller.dataConnection, 'SELECT * FROM CardTable')
```
Comment (fallinbryan) [inline source/developerHomeView.py]: getData takes one argument, a query..
`backlog = controller.dataConnection.getData(ScrumblesData.Query,getAllCards)`

File: source/developerHomeView.py [hunk 3]
```diff
@@ -15,12 +15,20 @@ def __init__(self, parent, controller,user):
...
-        assignedItem = ["Processing", "Statistics", "UML Diagram", "Use Cases", "Refactor", "User Stories"]
+        controller.dataConnection.connect()
+        backlog = ScrumblesData.ScrumblesData.getData(controller.dataConnection, 'SELECT * FROM CardTable')
+        backlog = [card['CardTitle'] for card in backlog]
+
+
+        teamMembers = ScrumblesData.ScrumblesData.getData(controller.dataConnection, 'SELECT UserName FROM UserTable')
+        teamMembers = [member['UserName'] for member in teamMembers]
+
+        assignedItem = ScrumblesData.ScrumblesData.getData(controller.dataConnection, 'SELECT * FROM CardTable')
```
Comment (fallinbryan) [inline source/developerHomeView.py]: In the next Sprint, the action of clicking a user will need to bound to a function that gets the items assigned to that user,

That is a query i dont have built into the query factory.
getCardsbyUserId(ScrublesObjects(user))

File: source/developerHomeView.py (hunk 2 above)
Comment (fallinbryan) [reply source/developerHomeView.py]: i see what you did there now.  you used getData as if it were a static method.  I didn't even know that was possible, lol.

I don't know if you checked out the [API](https://github.com/CEN3031-group16/GroupProject/wiki/Scrumbles-API-Documentation#query-class-definition) yet, but most queries we will need can be generated by the Query factory.

File: source/masterView.py [hunk 4]
```diff
@@ -90,6 +91,16 @@ def showCreateUserDialog(self):
         createUserDialog = Dialogs.CreateUserDialog(self,dataConnection)
         self.wait_window(createUserDialog.top)
 
+    def showCreateSprintDialog(self):
+        dbLoginInfo = ScrumblesData.DataBaseLoginInfo()
```
Comment (CEN3031-group16) [inline source/masterView.py]: can we get that persistent data connector in here? What was it, controller.dbConnector

**Pull request merged**